orderbook/src/database/fees.rs

```diff
@@ -0,0 +1,136 @@
+use super::Database;
+use crate::integer_conversions::*;
+
+use anyhow::{anyhow, Context, Result};
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Utc};
+use ethcontract::{H160, U256};
+use futures::StreamExt;
+
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub struct MinFeeMeasurement {
+    token: H160,
+    expiry: DateTime<Utc>,
+    min_fee: U256,
+}
+
+impl Database {
+    pub async fn save_fee_measurement(&self, measurement: MinFeeMeasurement) -> Result<()> {
+        const QUERY: &str =
+            "INSERT INTO min_fee_measurements (token, expiration_timestamp, min_fee) VALUES ($1, $2, $3);";
+        sqlx::query(QUERY)
+            .bind(measurement.token.as_bytes())
+            .bind(measurement.expiry)
+            .bind(u256_to_big_decimal(&measurement.min_fee))
+            .execute(&self.pool)
+            .await
+            .context("insert MinFeeMeasurement failed")
+            .map(|_| ())
+    }
+
+    pub async fn load_fee_measurements(
+        &self,
+        token: H160,
+        min_expiry: DateTime<Utc>,
+    ) -> Vec<MinFeeMeasurement> {
+        const QUERY: &str = "\
+            SELECT expiration_timestamp, min_fee FROM min_fee_measurements \
+            WHERE token = $1 AND expiration_timestamp >= $2
+            ";
```

Comment:
I think the way this used it should get the token and current time and return a single fee which is the minimum over all fees that are currently valid. That way no vector is returned.